### PR TITLE
Add networking requirements

### PR DESCRIPTION
**Pull request by pauloribeiro-codacy**
Added the networking requirements as identified in this issue:

https://codacy.atlassian.net/browse/DOCS-34

This was also identified while doing this user test:

https://codacy.atlassian.net/browse/CY-1170

### CODE REVIEW

File: docs/requirements.md [hunk 1]
```diff
@@ -11,13 +11,27 @@ The next sections describe in detail how to set up these prerequisites.
...
+|          | Service      | Protocol/Port | Notes                                                        |
+| -------- | ------------ | ------------- | ------------------------------------------------------------ |
+| Inbound  | SSH          | TCP/22        | Optional, to access the infrastructure remotely.             |
+| Inbound  | HTTP         | TCP/80        |                                                              |
+| Inbound  | HTTPS        | TCP/443       |                                                              |
+| Outbound | PostgreSQL   | TCP/5432      | Connection to the PostgreSQL DBMS                            |
+| Outbound | SMTP         | TCP/25        | Connection to your SMTP server                               |
+| Outbound | SMTPS        | TCP/465       | Connection to your SMTP server over TLS/SSL                  |
+| Outbound | Git provider | \*            | Connection to the ports required by your remote Git provider |
+
```
Comment (pauloribeiro-codacy) [inline docs/requirements.md:34]: I'm not sure if this information is enough for customers, or if we need to go into the detail of specifying which IP addresses / ranges are involved. :thinking: 

From my point of view we should have this simplified information for now and only add more detail if customers ask for it.

File: docs/requirements.md [hunk 2]
```diff
@@ -11,13 +11,27 @@ The next sections describe in detail how to set up these prerequisites.
...
 -   Tiller, the server part of [Helm](https://v2.helm.sh/) **version 2.16**, must be installed in the cluster
 -   The [NGINX Ingress controller](https://github.com/kubernetes/ingress-nginx) must be installed and correctly set up in the cluster
 
+### Cluster networking requirements
+
+The cluster must be configured to accept and establish connections on the following ports:
+
+|          | Service      | Protocol/Port | Notes                                                        |
+| -------- | ------------ | ------------- | ------------------------------------------------------------ |
+| Inbound  | SSH          | TCP/22        | Optional, to access the infrastructure remotely.             |
```
Comment (paulopontesm) [inline docs/requirements.md:27]: For "regular" kubernetes it doesn't make sense to put the 22 port. 
We only need kubectl correctly configured.

For microk8s we need the port 22.

File: docs/requirements.md [hunk 3]
```diff
@@ -11,13 +11,27 @@ The next sections describe in detail how to set up these prerequisites.
...
+| Inbound  | HTTP         | TCP/80        |                                                              |
+| Inbound  | HTTPS        | TCP/443       |                                                              |
+| Outbound | PostgreSQL   | TCP/5432      | Connection to the PostgreSQL DBMS                            |
+| Outbound | SMTP         | TCP/25        | Connection to your SMTP server                               |
+| Outbound | SMTPS        | TCP/465       | Connection to your SMTP server over TLS/SSL                  |
+| Outbound | Git provider | \*            | Connection to the ports required by your remote Git provider |
```
Comment (rtfpessoa) [inline docs/requirements.md:33]: Shouldn't we have dockerhub here too?

File: docs/requirements.md (hunk 2 above)
Comment (pauloribeiro-codacy) [reply docs/requirements.md:27]: Changing the notes to "**MicroK8s only**, to access the infrastructure remotely".

File: docs/requirements.md (hunk 3 above)
Comment (pauloribeiro-codacy) [reply docs/requirements.md:33]: We should. :+1: 

Unfortunately it seems it's not trivial to define the firewall rules to whitelist access to the Docker Hub. Docker ends up not providing the details (https://success.docker.com/article/is-there-a-whitelist-of-ips-used-by-docker-hub) and I also couldn't find a definitive answer online.

So I suggest that we add another row to the table with the aim of reminding the administrator about this and not so much to actually provide the values needed to configure a firewall.

Comment (rtfpessoa) [reply docs/requirements.md]: I think its two different problems. We never had ips in the configuration.
They could probably whitelist the dns instead, at least that was what we provided before.

Comment (rtfpessoa) [reply docs/requirements.md]: I tried to google about it and maybe it was never an option. Not sure why we had it there before.

**Pull request merged**